packages/ragbits-document-search/src/ragbits/document_search/ingestion/document_processor.py

```diff
@@ -106,7 +109,11 @@ def get_provider(self, document_meta: DocumentMeta) -> BaseProvider:
         Raises:
             ValueError: If no provider is found for the document type.
         """
-        provider = self._providers.get(document_meta.document_type)
-        if provider is None:
+        provider_class_or_provider = self._providers.get(document_meta.document_type)
+        if provider_class_or_provider is None:
             raise ValueError(f"No provider found for the document type {document_meta.document_type}")
+        elif isinstance(provider_class_or_provider, BaseProvider):
+            provider = provider_class_or_provider
+        else:
+            provider = provider_class_or_provider()
```

Comment:
maybe after first instantiation we should cache an instance within provider dict ??